### PR TITLE
fix :RedshiftDataOperator fails when return_sql_result is true

### PR DESCRIPTION
**Pull request by isatyamks**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/redshift_data.py [hunk 1]
```diff
@@ -162,9 +148,13 @@ def execute(self, context: Context) -> GetStatementResultResponseTypeDef | str:
                 )
 
         if self.return_sql_result:
-            result = self.hook.conn.get_statement_result(Id=self.statement_id)
-            self.log.debug("Statement result: %s", result)
-            return result
+            results = []
+            for i in range(1, len(self.sql) + 1):
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/operators/redshift_data.py]: What if `self.sql` is a string?

**Comment by isatyamks** [comment]
airflow/providers/amazon/aws/operators/redshift_data.py----->  
# Ensure self.sql is a list
    if isinstance(self.sql, str):
        self.sql = [self.sql]

File: airflow/providers/amazon/aws/operators/redshift_data.py [hunk 2]
```diff
@@ -110,63 +103,64 @@ def __init__(
...
-            self.log.warning(
-                "Invalid poll_interval:",
-                poll_interval,
-            )
+        self.poll_interval = max(poll_interval, 1)
         self.return_sql_result = return_sql_result
         self.statement_id: str | None = None
         self.deferrable = deferrable
 
     def execute(self, context: Context) -> GetStatementResultResponseTypeDef | str:
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/operators/redshift_data.py]: We now return a list right? Same for `execute_complete`

```suggestion
    def execute(self, context: Context) -> list[GetStatementResultResponseTypeDef] | str:
```

**Review by vincbeck — COMMENTED** [review]
Unit tests should also be updated and/or added

File: airflow/providers/amazon/aws/operators/redshift_data.py (hunk 2 above)
Comment (isatyamks) [reply airflow/providers/amazon/aws/operators/redshift_data.py]: yes

**Comment by isatyamks** [comment]
@vincbeck, @o-nikolas ,@eladkal please review the changes and proceed with the merge if the changes are accurate and appropriate.

**Comment by vincbeck** [comment]
> @vincbeck, @o-nikolas ,@eladkal please review the changes and proceed with the merge if the changes are accurate and appropriate.

Tests are failing, please fix them

**Comment by isatyamks** [comment]
@vincbeck  can you please help me to fix it

**Comment by vincbeck** [comment]
> @vincbeck can you please help me to fix it

Please look at the logs failure below. There are many. One example: https://github.com/apache/airflow/actions/runs/9765159007/job/26957199944?pr=40443

Have you run pre-commits? It seems you have not. A good start is to read the contributor quick start if you have not. https://github.com/apache/airflow/blob/main/contributing-docs/03_contributors_quick_start.rst

Other docs are definitely worth readable to fix your issues:
- Pre-commits (or static code checks): https://github.com/apache/airflow/blob/main/contributing-docs/08_static_code_checks.rst
- How to run tests locally: https://github.com/apache/airflow/blob/main/contributing-docs/09_testing.rst

**Comment by isatyamks** [comment]
its fixed!

**Pull request closed without merging**